2200007/괄호 짝짓기.py

```diff
@@ -0,0 +1,29 @@
+T = 10
+
+for t in range (1, 11) :
+    N = int(input())
+    str = input()
+    stack = []
+    val = 1
+
+    for n in range (N) :
+        if str[n] == '(' or str[n] == '{' or str[n] == '[' or str[n] == '<':
+            stack.append(str[n])
+        if str[n] == ')':
+            if len(stack) > 0 and stack.pop() != '(':
+                val = 0
+                break
+        if str[n] == '}':
+            if len(stack) > 0 and stack.pop() != '{':
+                val = 0
+                break
+        if str[n] == ']':
+            if len(stack) > 0 and stack.pop() != '[':
+                val = 0
+                break
+        if str[n] == '>':
+            if len(stack) > 0 and stack.pop() != '<':
+                val = 0
+                break
```

Comment:
깔끔한 풀이네요! 👍
테스트 케이스에서는 없었지만, 입력된 문자열에 닫는 괄호(`)`, `}`, `]`, `>`)가 먼저 나온다면, 0을 출력해야 합니다. 예를 들면,  아래와 같은 하나의 테스트 케이스에서 0이 출력되야 합니다.
```text
1
]
```
아래와 같이 수정해주시면 이 문제를 해결할 수 있습니다 😄 
```suggestion
        if str[n] == ')':
            if len(stack) == 0 or stack.pop() != '(':
                val = 0
                break
        if str[n] == '}':
            if len(stack) == 0 or stack.pop() != '{':
                val = 0
                break
        if str[n] == ']':
            if len(stack) == 0 or stack.pop() != '[':
                val = 0
                break
        if str[n] == '>':
            if len(stack) == 0 or stack.pop() != '<':
                val = 0
                break
```